### PR TITLE
adios2: Fix build with gcc@10:

### PR DESCRIPTION
**Pull request by michaelkuhn**

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@ax3l @chuckatkins @williamfgc

**Comment by ax3l** [comment]
Thank you! :rocket: :sparkles: 

This is likely only needed for the Fortran bindings and likely an actual bug. Let's quickly wait for feedback from upstream: https://github.com/ornladios/ADIOS2/issues/2228

File: var/spack/repos/builtin/packages/adios2/package.py [hunk 1]
```diff
@@ -118,6 +118,10 @@ class Adios2(CMakePackage):
     # See https://github.com/ornladios/ADIOS2/pull/1899
     patch('2.5-fix-clear_cache.patch', when='@2.5.0')
 
+    def setup_build_environment(self, env):
+        if self.spec.satisfies('%gcc@10:'):
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/adios2/package.py]:
```suggestion
        if self.spec.satisfies('%gcc@10: +fortran'):
```

File: var/spack/repos/builtin/packages/adios2/package.py [hunk 2]
```diff
@@ -118,6 +118,10 @@ class Adios2(CMakePackage):
     # See https://github.com/ornladios/ADIOS2/pull/1899
     patch('2.5-fix-clear_cache.patch', when='@2.5.0')
 
+    def setup_build_environment(self, env):
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/adios2/package.py:121]: I opened an upstream issue that we can reference here:
```suggestion
    def setup_build_environment(self, env):
        # https://github.com/ornladios/ADIOS2/issues/2228
```

**Review by alalazo — APPROVED** [review]

**Review by chuckatkins — APPROVED** [review]
This is appropriate for the current release.  If it's changed in the upcoming 2.6 release then we can just adjust the conditional accordingly.

**Comment by ax3l** [comment]
Awesome, thanks for the feedback @chuckatkins !

**Pull request merged**